### PR TITLE
[`Moe`] Enable aux loss automatically when in training + coef is not 0

### PR DESCRIPTION
**Pull request by vasqu**
As per title, WIP --> needs a test

### CODE REVIEW

File: src/transformers/utils/generic.py [hunk 1]
```diff
@@ -906,6 +906,16 @@ def wrapper(self, *args, **kwargs):
...
 
+        # Maybe temporarily enable collecting router logits for aux loss during training
+        original_output_router_logits = None
+        if (
+            self.training
+            and (router_aux_loss_coef := getattr(self.config, "router_aux_loss_coef", None))
+            and router_aux_loss_coef != 0
+        ):
+            original_output_router_logits = self.config.output_router_logits
+            self.config.output_router_logits = True
```
Comment (vasqu) [inline src/transformers/utils/generic.py:917]: Note: Conditioned on training because otherwise we would always output the logits

**Review by redpanda1995 — APPROVED** [review]

File: src/transformers/utils/generic.py (hunk 1 above)
Comment (ArthurZucker) [reply src/transformers/utils/generic.py:917]: it makes sense tho I don't remember what's the argument for people to not pass `output_router_logits`? its not intuitive?

**Review by ArthurZucker — APPROVED** [review]
lgtm just let's document why we are enabling this

File: src/transformers/utils/generic.py (hunk 1 above)
Comment (vasqu) [reply src/transformers/utils/generic.py:917]: Honestly not sure, it's based on the issue from #44242 and seemingly @Rocketknight1 seeing more of these; so seems convenient enough. Will check with the user where he found it in the docs tho :eyes:

**Comment by winglian** [comment]
removing `output_router_logits` is likely going to break things downstream, most notably [Liger-Kernel](https://github.com/linkedin/Liger-Kernel/blob/5057ca8799f2ad2d7e36a1c1440a68f4f037f27b/src/liger_kernel/transformers/model/gpt_oss.py#L26),

**Comment by vasqu** [comment]
It is hidden behind the decorator, not removed. Similar to `output_hidden_states`
